src/langauge-servers/javaUtils.ts

```diff
@@ -0,0 +1,187 @@
+import { getLogger } from './../logger';
+import * as path from 'path'
+import * as fs from 'fs'
+import * as cp from 'child_process'
+import * as os from 'os'
+import _ = require('lodash')
+
+const isWindows = process.platform.match('win32')
+const isMac = process.platform.match('darwin')
+const isLinux = process.platform.match('linux')
+export const JAVA_FILENAME = 'java' + ((process.platform === 'win32') ? '.exe' : '');
+
+export interface JavaRuntime {
+    location: string;
+    version: number;
+}
+
+/**
+ * return metadata for all installed JDKs.
+ */
+export const findJavaHomes = (): JavaRuntime[] => {
+    const javaBinLocations: JavaRuntime[] = [];
+    const jdkSet = new Set<string>();
+
+    updateJDKs(jdkSet, getJavaLocationsfromEnv("JDK_HOME"));
+    updateJDKs(jdkSet, getJavaLocationsfromEnv("JAVA_HOME"));
+    updateJDKs(jdkSet, fromCommonPlaces());
+
+    jdkSet.forEach(jdkLocation => {
+        getLogger('langserver').debug('Java - found location found', { jdkLocation })
+        const javaBin = path.join(jdkLocation, "bin", JAVA_FILENAME)
+
+        if (fs.existsSync(javaBin)){
+            const version = getJavaVersion(jdkLocation);
+
+            if (version) {
+                javaBinLocations.push({
+                    location: jdkLocation,
+                    version
+                });
+            } else {
+                getLogger("langserver").warn("Java - no java exec was found")
+            }
+        }
+    })
+    return javaBinLocations;
+}
+
+const updateJDKs = (set: Set<string>, newJdks: string[]) => {
+    newJdks.forEach(jdkLoc => {set.add(jdkLoc); console.log(jdkLoc)})
+}
+
+const getJavaLocationsfromEnv = (name: string): string[] => {
+    if (!process.env[name]) {
+        return []
+    }
+
+    const workspaces = process.env[name].split(path.delimiter);
```

Comment:
`workspace -> folders`?  
Maybe it's also worth mentioning the format of the variable your'e expecting here.  
```
// e.g: '/path/to/something1:/path/to/some/other/crap'
```